source/common/ssl/context_config_impl.cc

```diff
@@ -34,6 +34,61 @@ const std::string ContextConfigImpl::DEFAULT_CIPHER_SUITES =
 
 const std::string ContextConfigImpl::DEFAULT_ECDH_CURVES = "X25519:P-256";
 
+static std::vector<bssl::UniquePtr<X509_CRL>>
+loadCRL(const envoy::api::v2::CertificateValidationContext& vctx) {
+  std::vector<bssl::UniquePtr<X509_CRL>> crls;
+
+  if (!vctx.has_crl()) {
+    return crls;
+  }
+
+  // We want to load multiple CRLs from the same file, so we need to use
+  // OpenSSL's STACK abstraction. We read from the input file or inline data,
+  // and load into the stack of X509_INFOs.
+  bssl::UniquePtr<STACK_OF(X509_INFO)> xis(nullptr);
+
+  auto crl_conf = vctx.crl();
+  if (crl_conf.filename().length()) {
```

Comment:
You should probably wait for `ContextConfigImpl::readDataSource()` from https://github.com/envoyproxy/envoy/pull/2248.

Reply:
+1

If not though, change to "if (!crl_conf.filename().empty()) {"